sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -261,23 +262,71 @@ public final BlobInputStream openInputStream(BlobRange range, BlobRequestConditi
      */
     public BlobInputStream openInputStream(BlobInputStreamOptions options) {
         options = options == null ? new BlobInputStreamOptions() : options;
+        ConcurrencyControl concurrencyControl = options.getConcurrencyControl() == null ? ConcurrencyControl.ETAG
+            : options.getConcurrencyControl();
 
-        BlobProperties properties = getProperties();
+        BlobProperties properties = getPropertiesWithResponse(options.getRequestConditions(), null, null).getValue();
+        String eTag = properties.getETag();
+        String versionId = properties.getVersionId();
 
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
         BlobRequestConditions requestConditions = options.getRequestConditions() == null
             ? new BlobRequestConditions() : options.getRequestConditions();
-        // Target the user specified version by default. If not provided, target the latest version.
-        if (requestConditions.getIfMatch() == null) {
-            requestConditions.setIfMatch(properties.getETag());
+        BlobAsyncClientBase client = this.client;
+
+        switch (concurrencyControl) {
+            case NONE:
+                if (requestConditions.getIfMatch() != null) {
+                    throw logger.logExceptionAsError(generateControlException("requestConditions.ifMatch",
+                        ConcurrencyControl.NONE.toString(), ConcurrencyControl.ETAG.toString()));
+                }
+                if (this.client.getVersionId() != null) {
+                    throw logger.logExceptionAsError(generateControlException("client.versionId",
+                        ConcurrencyControl.NONE.toString(), ConcurrencyControl.VERSION_ID.toString()));
+                }
+                break;
+            case ETAG:
+                if (this.client.getVersionId() != null) {
+                    throw logger.logExceptionAsError(generateControlException("client.versionId",
+                        ConcurrencyControl.ETAG.toString(), ConcurrencyControl.VERSION_ID.toString()));
+                }
+                // Target the user specified eTag by default. If not provided, target the latest eTag.
+                if (requestConditions.getIfMatch() == null) {
+                    requestConditions.setIfMatch(eTag);
+                }
+                break;
+            case VERSION_ID:
+                if (requestConditions.getIfMatch() != null) {
+                    throw logger.logExceptionAsError(generateControlException("requestConditions.ifMatch",
+                        ConcurrencyControl.VERSION_ID.toString(), ConcurrencyControl.ETAG.toString()));
+                }
```

Comment:
How about other requestConditions?

Reply:
I think only etag is checked because we only lock on the etag. The other request conditions are only used in the initial get properties and then will be replaced by the etag (if that's the control type specified). Although I can see how it might be odd to specify LMT and version at the same time....

Reply:
Maybe they just want that extra level of assurance :P